qualtran/cirq_interop/decompose_protocol.py

```diff
@@ -0,0 +1,98 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Any, FrozenSet, Sequence
+
+import cirq
+from cirq.protocols.decompose_protocol import DecomposeResult
+
+_FREDKIN_GATESET = cirq.Gateset(cirq.FREDKIN, unroll_circuit_op=False)
+
+
+def _fredkin(qubits: Sequence[cirq.Qid], context: cirq.DecompositionContext) -> cirq.OP_TREE:
```

Comment:
reminder that per our agreement if this code still exists like 1 month from now I will unceremoniously remove it

Reply:
I don't want to keep this code either. The project that takes care of supporting different decompositions when counting resources should remove this code as a first proof of concept. Whoever removes this code should make sure the T-counts of higher level primitives remain the same, and then I'm happy.